### DISCUSSION TITLE
Xunit - Add Console Log to Report for Failed Tests.

### DISCUSSION THREAD

**Discussion by dimabaskin** [allure-framework/allure2#2178] [Questions & Support]
HI . I'm started using Allure report in Test project based C# Selenium XUnit.
I'm Using Nlog as a logger . And one of loger target is Console.
So for Tests in Status Passed allure generated json with console log in "message" parameter.
But for Test in Status Failed :
"message" parameter contains Assert message. (from FluentAssertion )
"trace" parameter contains stack trace .
How Can I Add a Console log to Failed Tests ?

Failed Test have 3 section in tetslog  file
Message
Stack Trace
Standard Output

**Comment by ZeddzZ** [comment]
@dimabaskin 
Hi! Have you figured out how to send log for failed tests? I faced the same issue, but can't find any appropriate solution.

**Reply by ZeddzZ** [reply]
@baev, @delatrie
Hi guys, sorry for tagging you, but maybe you know how to overcome this situation?
 Here is how it looks like:
Allure Test ops, Same message is in allure-results folder
![image](https://github.com/allure-framework/allure2/assets/11161671/d41add68-c5ba-405c-ae96-933119af72d4)
Testresults file that xUnit generated
![image](https://github.com/allure-framework/allure2/assets/11161671/e0e7eb05-6a10-4ebf-a6bb-7b8e4caa5623)

One of the reasons of using Allure was monitoring failed tests and by its data find issues, but without log we can do nothing to fix such flacky tests.
Thanks for possible help!

**Reply by dimabaskin** [reply]
Hi. Yes. By using allure attachment. In nlog configuration you can set to save each test log as separate file and in case test failed add it as attachment to report.
